### PR TITLE
Support List<Object> properties in swift migrations

### PR DESCRIPTION
**Pull request by alazier**
@segiddins @jpsim

### CODE REVIEW

File: Realm/RLMObjectSchema.mm [hunk 1]
```diff
@@ -118,7 +118,9 @@ + (instancetype)schemaForObjectClass:(Class)objectClass {
     if (NSString *primaryKey = [objectClass primaryKey]) {
         for (RLMProperty *prop in schema.properties) {
             if ([primaryKey isEqualToString:prop.name]) {
-                prop.indexed = YES;
+                if (prop.type == RLMPropertyTypeString) {
```
Comment (segiddins) [inline Realm/RLMObjectSchema.mm]: is this change related, or is this for the int index issues?

File: Realm/RLMResults_Private.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+#import "RLMResults.h"
+
+@class RLMObjectSchema;
+
+@interface RLMResults () {
+  @public
+    RLMObjectSchema *_objectSchema;
+}
+@end
```
Comment (segiddins) [inline Realm/RLMResults_Private.h]: newline

File: examples/ios/swift/RealmExamples.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -754,6 +748,11 @@
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				ASSETCATALOG_COMPILER_LAUNCHIMAGE_NAME = LaunchImage;
+				FRAMEWORK_SEARCH_PATHS = (
```
Comment (segiddins) [inline examples/ios/swift/RealmExamples.xcodeproj/project.pbxproj]: Not every developer is named `ari`

File: Realm/RLMObjectSchema.mm (hunk 1 above)
Comment (jpsim) [inline Realm/RLMObjectSchema.mm]: This seems wrong.

File: Realm/RealmSwift/Migration.swift [hunk 4]
```diff
@@ -57,3 +58,42 @@ public class Migration {
...
+public class MigrationObject : Object {
+    subscript(key: String) -> AnyObject? {
+        if (self.objectSchema[key].type == RLMPropertyType.Array) {
+            return listProperties[key]
+        }
+        return super[key]
+    }
+
+    public func initalizeListPropertyWithName(name: String, rlmArray: RLMArray) {
+        var list = List<Object>()
```
Comment (jpsim) [inline Realm/RealmSwift/Migration.swift]: Couldn't these 3 lines could be replaced with `listProperties[name] = List<Object>(rlmArray)`?

File: examples/ios/swift/RealmExamples.xcodeproj/project.pbxproj (hunk 3 above)
Comment (jpsim) [inline examples/ios/swift/RealmExamples.xcodeproj/project.pbxproj]: I think this whole file should be reverted back to its previous version.

File: Realm/RLMObjectSchema.mm (hunk 1 above)
Comment (alazier) [inline Realm/RLMObjectSchema.mm]: This is correct. Int indexing does not work. It was reverted on master. This reflects those changes on this branch

File: examples/ios/swift/RealmExamples.xcodeproj/project.pbxproj (hunk 3 above)
Comment (alazier) [inline examples/ios/swift/RealmExamples.xcodeproj/project.pbxproj]: done

File: Realm/RealmSwift/Migration.swift (hunk 4 above)
Comment (alazier) [inline Realm/RealmSwift/Migration.swift]: ok

**Comment by jpsim** [comment]
:+1:
